sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala

```diff
@@ -2476,11 +2703,10 @@ class AdaptiveQueryExecSuite
             "UNION ALL SELECT key2 FROM skewData2 GROUP BY key2", 1, 1)
 
         // skewJoin1 union (skewJoin2 join aggregate)
-        // skewJoin2 will lead to extra shuffles, but skew1 cannot be optimized
          checkSkewJoin(
           "SELECT key1 FROM skewData1 JOIN skewData2 ON key1 = key2 UNION ALL " +
             "SELECT key1 from (SELECT key1 FROM skewData1 JOIN skewData2 ON key1 = key2) tmp1 " +
-            "JOIN (SELECT key2 FROM skewData2 GROUP BY key2) tmp2 ON key1 = key2", 3, 0)
+            "JOIN (SELECT key2 FROM skewData2 GROUP BY key2) tmp2 ON key1 = key2", 3, 3)
```

Comment:
this newly added skew test case can be optimized by this PR without extra shuffle:

Master | this PR
--- | ---
![master_skew_case](https://user-images.githubusercontent.com/7322292/153830682-e83ac367-a4d6-4bcf-8699-bd5a944f70e2.png) | ![general_skew_case](https://user-images.githubusercontent.com/7322292/153830711-bb7d1bdf-e59e-400d-a79d-a1e61c9ec8e5.png)